edalize/surelog.py

```diff
@@ -0,0 +1,59 @@
+import logging
+import os.path
+
+from edalize.edatool import Edatool
+
+logger = logging.getLogger(__name__)
+
+class Surelog(Edatool):
+
+    argtypes = ['vlogdefine', 'vlogparam']
+
+    @classmethod
+    def get_doc(cls, api_ver):
+        if api_ver == 0:
+            return {'description' : "Surelog",
+                    'lists' : [
+                        {'name' : 'library_files',
```

Comment:
This needs som explanation. I get the feeling it is verilog "libraries", i.e. a directory with files that have the same filename as the modules inside, or is it something else? If it is verilog libraries I wonder if this is really the method we prefer. It should be possible to specify these as ordinary source files instead.

Generally when files are added as tool_options we must be very careful about thinking how the user will pass them into Edalize. In most cases it's better to add a new file type if we must

Reply:
Surelog requires access to definition of all modules to match the usage with the model specification (without this, Surelog will be missing parameter information for those modules).

This parameter allows to pass such files as a libraries (more information available in: https://github.com/alainmarcel/Surelog/issues/893).

I tried to pass them as a ``logical_name`` file_type , but when using ``edalize`` with ``fusesoc``, files can only be specified relative to the ``.core`` location (and this files are usually stored outside of that location). If there is better way to pass this files that will work with ``fusesoc`` I will change this.

Example of such library file: https://github.com/antmicro/yosys/blob/uhdm-yosys/techlibs/xilinx/cells_xtra_surelog.v

Reply:
They are only needed, when a user uses a Xilinx primitive in their code.

Reply:
Potentially any vendor primitives (this is not limited to Xilinx)

Reply:
OK, a follow-up then: How are those libraries distributed? Are they part of Surelog, or of (the modified version of) Yosys? Or is the user expected to provide those files themselves?

Reply:
They are distributed as a part of ``yosys`` or ``symbiflow_toolchain`` (in case of building using ``symbiflow_toolchain`` instead of ``vivado``).